src/Components/NewProfile/NewProfile.js

```diff
@@ -1,84 +1,66 @@
-import React from 'react';
-import { connect } from 'react-redux';
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { updateUser } from "../../Redux/User/userSlice.js";
+import "./NewProfile.styles.scss";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import {
-  setFirstName,
-  setLastName,
-  setEmail,
-  clearForm
-} from '../../Redux/Actions/actions.js';
+  faEnvelope,
+  faPhone,
+  faUser,
+} from "@fortawesome/free-solid-svg-icons";
+import { useTranslation } from "react-i18next";
 
-const ProfileForm = (props) => {
-  const {
-    firstName,
-    lastName,
-    email,
-    setFirstName,
-    setLastName,
-    setEmail,
-    clearForm
-  } = props;
+
+const NewProfile = () => {
+  const dispatch = useDispatch();
+  const user = useSelector((state) => state.reducer.user);
+  const { t } = useTranslation();
+  const [username, setUsername] = useState(user.username);
+  const [email, setEmail] = useState(user.email);
+  const [phoneNumber, setPhoneNumber] = useState(user.phoneNumber);
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    console.log({
-      firstName,
-      lastName,
-      email
-    });
-    clearForm();
+    dispatch(updateUser({ username, email, phoneNumber }));
+    alert("Profile updated successfully!");
```

Comment:
Please add a comment `TODO: move notifications to redux-saga` we will take care of it later

Reply:
Interesting I did not know that this was a feature that highlights TODO comments. I appreciate the tip.